Source/WebCore/platform/sql/SQLiteDatabase.h

```diff
@@ -49,7 +50,7 @@ class DatabaseAuthorizer;
 class SQLiteStatement;
 class SQLiteTransaction;
 
-class SQLiteDatabase : public CanMakeThreadSafeCheckedPtr {
+class SQLiteDatabase : public CanMakeWeakPtr<SQLiteDatabase>, public CanMakeThreadSafeCheckedPtr {
```

Comment:
Can we remove `CanMakeThreadSafeCheckedPtr` or are there still places it is used?

Don’t we need a thread-safe `WeakPtr` as well?

Reply:
CheckedPtr is still in use for pointers on the stack so I can’t remove it.

WeakPtr already has thread safe ref count by default, SingleThreadWeakPtr is the one that doesn’t.